### PR TITLE
Updated json Object creation. Only one TransactionObject with arraylists

### PR DESCRIPTION
**Pull request by moekappels**

### CODE REVIEW

**Comment by kiview** [comment]
Description is always appreciated...

File: src/main/java/de/internetsicherheit/brl/bloxberg/cache/ethereum/ReducedTransObjectArray.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import org.web3j.protocol.core.methods.response.Transaction;
+
+import java.util.ArrayList;
+
+public class ReducedTransObjectArray {
+
+    /**
+     * a simple custom Datatype that represents a combination of a blocknumber and the corresponding transactioncount
+          */
+    public ArrayList<String> fromAddress;
```
Comment (kiview) [inline src/main/java/de/internetsicherheit/brl/bloxberg/cache/ethereum/ReducedTransObjectArray.java:12]:
```suggestion
    public final List<String> fromAddress = new ArrayList<>();
```
 If we make fields public, let's at least make the reference immutable. Also, use interface type and diamond operator.

File: src/main/java/de/internetsicherheit/brl/bloxberg/cache/ethereum/ReducedTransObjectArray.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    public ArrayList<String> toAddress;
+
+    public ReducedTransObjectArray() {
+
+        this.fromAddress = new ArrayList<String>();
+        this.toAddress = new ArrayList<String>();
+
+    }
```
Comment (kiview) [inline src/main/java/de/internetsicherheit/brl/bloxberg/cache/ethereum/ReducedTransObjectArray.java:20]:
```suggestion
```
Default constructor unnecessary.

File: src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/TestCenter.java [hunk 3]
```diff
@@ -32,17 +30,19 @@ public TestCenter(BloxbergClient client, File outputfile) {
 
     public void generateJsonFile(int start, int stop) throws IOException {
 
-        FileWriter fileWriter = new FileWriter(outputfile, true);
+        //FileWriter fileWriter = new FileWriter(outputfile, true);
```
Comment (kiview) [inline src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/TestCenter.java:33]: Please remove all commented lines.

File: src/main/java/de/internetsicherheit/brl/bloxberg/cache/ethereum/ReducedTransObjectArray.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+import java.util.ArrayList;
+
+public class ReducedTransObjectArray {
+
+    /**
+     * a simple custom Datatype that represents a combination of a blocknumber and the corresponding transactioncount
+          */
+    public ArrayList<String> fromAddress;
+    public ArrayList<String> toAddress;
```
Comment (kiview) [inline src/main/java/de/internetsicherheit/brl/bloxberg/cache/ethereum/ReducedTransObjectArray.java:13]:
```suggestion
    public final List<String> toAddress = new ArrayList<>();;
```

File: src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/HistoricDataVisualizer.java [hunk 5]
```diff
@@ -62,8 +62,8 @@ public void start(Stage primaryStage) throws IOException {
         // original: writer = new EthereumWriter(Path.of(OUTPUTDIRECTORY), filename);
         writer = new EthereumWriter(Path.of(OUTPUTDIRECTORY), filename);
         BlockAggregator dbs = initDataBlockSummerizer(filename);
-        TestCenter testCenter = new TestCenter(client, new File(OUTPUTDIRECTORY + "extractedData6.json"));
-        testCenter.generateJsonFile(0, 10000);
+        TestCenter testCenter = new TestCenter(client, new File(OUTPUTDIRECTORY + "transactionsFrom100kBlocks.json"));
+        testCenter.generateJsonFile(0, 100000);
```
Comment (kiview) [inline src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/HistoricDataVisualizer.java:66]: Why are these changes necessary in this PR?

File: src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/TestCenter.java [hunk 6]
```diff
@@ -32,17 +30,19 @@ public TestCenter(BloxbergClient client, File outputfile) {
...
+            extractJsonObject(start, rdoa);
         }
-        seqWriter.close();;
+        //seqWriter.close();;
+        objectMapper.writeValue(outputfile, rdoa);
 
     }
 
-    public void extractJsonObject(int blockNumber, SequenceWriter seqWriter) throws IOException {
+    public void extractJsonObject(int blockNumber, ReducedTransObjectArray rdoa) throws IOException {
```
Comment (kiview) [inline src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/TestCenter.java:45]: Don't make a void method and then mutate a parameter. Instead, use return values.

File: src/main/java/de/internetsicherheit/brl/bloxberg/cache/ethereum/ReducedTransObjectArray.java (hunk 1 above)
Comment (moekappels) [reply src/main/java/de/internetsicherheit/brl/bloxberg/cache/ethereum/ReducedTransObjectArray.java:12]: ok, so I think I understand everything so far, but what does the empty diamond operator do? generics require a type, right?

File: src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/HistoricDataVisualizer.java (hunk 5 above)
Comment (moekappels) [reply src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/HistoricDataVisualizer.java:66]: they are not absolutely necessary, but i like to change filenames frequently so I dont accidentally overwrite files.

File: src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/TestCenter.java (hunk 6 above)
Comment (moekappels) [reply src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/TestCenter.java:45]: but it s still ok to mutate a parameter and use that parameter then as a return value, right?

File: src/main/java/de/internetsicherheit/brl/bloxberg/cache/ethereum/ReducedTransObjectArray.java (hunk 1 above)
Comment (kiview) [reply src/main/java/de/internetsicherheit/brl/bloxberg/cache/ethereum/ReducedTransObjectArray.java:12]: Thanks to type inference, Java SE 7 and above allow the programmer to substitute an empty pair of angle brackets (<>, called the diamond operator) for a pair of angle brackets containing the one or more type parameters that a sufficiently-close context implies.

So said in simple words: If the type on the left already tells you the generic type, the compiler can infer it and it is not necessary it specify it on the right as well.

File: src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/HistoricDataVisualizer.java (hunk 5 above)
Comment (kiview) [reply src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/HistoricDataVisualizer.java:66]: While I understand this, this means those filenames should not be hardcoded in the source code. So have them as arguments provided to your program or, at least, don't commit them, as this is creating noise in the git history.

File: src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/TestCenter.java (hunk 6 above)
Comment (kiview) [reply src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/TestCenter.java:45]: No, this is terrible style and not idiomatic Java. 
If you want to mutate values, mutate your fields (or create completely new objects). Follow OOP encapsulation principles.

**Comment by kiview** [comment]
@moekappels Which former PR? I don't see any open PRs.

**Comment by moekappels** [comment]
meant this one. the changes create a file structure that is preshaped. which is not what you wanted. so the whole pull request should just be closed.
